### PR TITLE
Release/v0.13.0

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #70 
- #77 

**This PR will result in the following new package version:**

- Breaking since the folder `stripe_reports` was renamed to `stripe_financial_reports` and could break if users relied on the old name.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Changes 🚨
> - Renamed folder `stripe_reports` to `stripe_financial_reports` to be more descriptive of the contents.
>   - ⚠️ If you are using folder names to scope out configs, runs, etc., you will need to update the folder name.
> 
> ## Bug fixes
> - Updated model `int_stripe__date_spine` to accommodate cases when model `stripe__balance_transactions` has no records. 
>   - Previously, the date spine relied on `stripe__balance_transactions` to set date bounds, which caused errors if empty. Now, the model defaults to a one-month range in such cases.
> 
> ## Under the hood
> - Updated structure of model `int_stripe__date_spine` for improved performance and maintainability.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
  - Will regen after approval.
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- Validation tests pass.

### If you had to summarize this PR in an emoji, which would it be?
 
🍓

### CODE REVIEW

File: integration_tests/tests/integrity/integrity_daily_overview.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+daily_overview as (
+    select count(*) as daily_overview_count
+    from {{ target.schema }}_stripe_dev.stripe__daily_overview
+)
+
+-- test will return values and fail if the row counts don't match
+select *
+from spine
+join daily_overview
+    on spine.spine_count != daily_overview.daily_overview_count
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/integrity/integrity_daily_overview.sql:21]: Disregard this comment. I couldn't delete it for some reason.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+# dbt_stripe v0.13.0
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: Be sure to also update the version in the README

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz thanks for working on this update! I just have a few comments and suggestions before approving. Let me know if you have any questions regarding my comments. Thanks!

File: models/intermediate/int_stripe__date_spine.sql [hunk 3]
```diff
@@ -1,62 +1,43 @@
...
     {% set last_date_query %}
-        select  max( balance_transaction_created_at ) as max_date from {{ ref('stripe__balance_transactions') }}
-    {% endset %}
-
-    {% set current_date_query %}
-        select current_date
+        select coalesce(
+            greatest(max(cast(balance_transaction_created_at as date)), cast(current_date as date)),
+            cast(current_date as date)
+            ) as max_date
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/intermediate/int_stripe__date_spine.sql:20]: The greatest is to imitate the prior code's line 30. I did take into account that the greatest of a date and null is a null, so that's why there is the coalesce of line 17. If the greatest check produces a null, then it uses the current date. I also made sure to test this by supplying empty source tables.

File: models/intermediate/int_stripe__date_spine.sql [hunk 4]
```diff
@@ -1,62 +1,43 @@
...
-            {% set first_date_adjust = "cast('" ~ first_date[0:10] ~ "' as date)" %}
 
-        {% else %}
-            {% set first_date_adjust = "'" ~ first_date[0:10] ~ "'" %}
+    {%- set first_date_query %}
+        select coalesce(
+            min(cast(balance_transaction_created_at as date)), 
+            cast({{ dbt.dateadd("month", -1, "current_date") }} as date)
+            ) as min_date
+        from {{ ref('stripe__balance_transactions') }}
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/intermediate/int_stripe__date_spine.sql:11]: I didn't this time and followed what was used in the current version since I know in the past when we've done that, someone that was using a custom source had an issue. However, I agree that would make things easier. I am open to use either ref or the source.

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:1]: Sorry about that. Thanks!

File: models/intermediate/int_stripe__date_spine.sql [hunk 5]
```diff
@@ -1,62 +1,43 @@
...
-        select  max( balance_transaction_created_at ) as max_date from {{ ref('stripe__balance_transactions') }}
-    {% endset %}
-
-    {% set current_date_query %}
-        select current_date
+        select coalesce(
+            greatest(max(cast(balance_transaction_created_at as date)), cast(current_date as date)),
+            cast(current_date as date)
+            ) as max_date
+        from {{ ref('stripe__balance_transactions') }}
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/intermediate/int_stripe__date_spine.sql:21]: Same consideration as above!

File: models/intermediate/int_stripe__date_spine.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_stripe__date_spine.sql:11]: You make a good point, and I am unsure what this impact would be for folks using the union feature. It would just pick one source and not take into account all other transactions across the other sources. With that, I say we keep the ref as you have it.

File: models/intermediate/int_stripe__date_spine.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_stripe__date_spine.sql:20]: Ah yeah I see what you are saying and see how you accounted for this. Great, thanks for sharing the detail behind your code update. Looks good to me!

File: models/intermediate/int_stripe__date_spine.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_stripe__date_spine.sql:21]: No changes required base on the above comment

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-catfritz thanks for answering my questions! Given the conversations from the comments, this looks good to go!

**Review by fivetran-reneeli — APPROVED** [review]
Looks good!
